felix/fv/ep_to_host_action_test.go

```diff
@@ -0,0 +1,236 @@
+				f.Exec("iptables", "-t", "filter", "-P", "INPUT", chainPolicy)
+			}
+			// Set Felix policy.
+			By("Updating DefaultEndpointToHostAction")
+			utils.UpdateFelixConfig(client, func(configuration *api.FelixConfiguration) {
+				configuration.Spec.DefaultEndpointToHostAction = epToHostPol
+			})
+
+			switch hepIface {
+			case "none":
+			case "eth0", "*", "*-pre-dnat":
+				policy := api.NewGlobalNetworkPolicy()
+				policy.Name = "default-hep-egress"
+				policy.Spec.Selector = "hep=='true'"
+				policy.Spec.Egress = []api.Rule{
+					{
+						Action: api.Allow,
+					},
+				}
+				_, err := client.GlobalNetworkPolicies().Create(utils.Ctx, policy, utils.NoOptions)
+				Expect(err).NotTo(HaveOccurred())
+
+				policy = api.NewGlobalNetworkPolicy()
+				policy.Name = "default-hep"
+				policy.Spec.Selector = "hep=='true'"
+
+				if hepIface == "*-pre-dnat" {
+					policy.Spec.ApplyOnForward = true
+					policy.Spec.PreDNAT = true
+					hepIface = "*"
+
+					// pre-DNAT applies to forwarded traffic, need to allow workload traffic explicitly.
+					policy.Spec.Ingress = append(policy.Spec.Ingress, api.Rule{
+						Action: api.Allow,
+						Source: api.EntityRule{
+							Selector: "!has(hep)",
+						},
+						Destination: api.EntityRule{
+							Selector: "!has(hep)",
+						},
+					})
+				}
+
+				policy.Spec.Ingress = append(policy.Spec.Ingress, api.Rule{
+					Action: hepPolicy,
+					Source: api.EntityRule{
+						Selector: "!has(hep)",
+					},
+				})
+
+				_, err = client.GlobalNetworkPolicies().Create(utils.Ctx, policy, utils.NoOptions)
+				Expect(err).NotTo(HaveOccurred())
+
+				for _, f := range tc.Felixes {
+					hep := api.NewHostEndpoint()
+					hep.Name = "hep-" + f.Name
+					hep.Labels = map[string]string{
+						"hep": "true",
+					}
+					hep.Spec.Node = f.Hostname
+					hep.Spec.ExpectedIPs = []string{f.IP}
+					hep.Spec.InterfaceName = hepIface
+					_, err := client.HostEndpoints().Create(context.TODO(), hep, options.SetOptions{})
+					Expect(err).NotTo(HaveOccurred())
+				}
+			default:
+				Fail("Unknown hep type: " + hepIface)
+			}
+
+			// Check that we start with no WEP-to-WEP connectivity.  This ensures that, when
+			// we see WEP-to-WEP connectivity in the check below, we know that Felix has
+			// finished programming the dataplane.
+			cc.Expect(connectivity.None, w[0], w[1])
+			cc.CheckConnectivity()
+			cc.ResetExpectations()
+
+			// Trigger startup, Felix will apply the ep-to-host policy and the WEP-to-WEP
+			// allow profile in one transaction.
+			for _, f := range tc.Felixes {
+				f.TriggerDelayedStart()
+			}
+
+			// Now add the default allow profile, which should give us WEP-to-WEP connectivity.
+			// When we get WEp-to-WEP, we know that Felix has finished programming so the
+			// WEP-to-host test is valid.
+
+			By("Checking connectivity")
+			cc.Expect(expectedConn, w[0], hostW[0])
+			cc.Expect(expectedConn, w[1], hostW[1])
```

Comment:
I'd expect local and remote pod to have possibly different connectivity. If the epToHost == DROP, the remote pod should still connect if other policies permit, right?

Reply:
They're each connecting to their own host, so should be symmetric?